test/source/tests/tests/gmail.ts

```diff
@@ -116,6 +116,19 @@ export const defineGmailTests = (testVariant: TestVariant, testWithBrowser: Test
       const composePage = await GmailPageRecipe.openSecureCompose(t, gmailPage, browser);
     }));
 
+    ava.default.only('mail.google.com - decrypt message in offline mode', testWithBrowser('compatibility', async (t, browser) => {
+      const gmailPage = await BrowserRecipe.openGmailPage(t, browser);
+      await gmailPage.type('[aria-label="Search mail"]', 'encrypted + signed with gpg');
+      await gmailPage.press('Enter'); // submit search
+      await gmailPage.page.waitFor(1000); // wait for search results
+      await gmailPage.page.setOfflineMode(true); // go OFFLINE
+      await gmailPage.press('Enter'); // open the message
+      // getFramesUrls will return one iframe which is expected
+      const urls = await gmailPage.getFramesUrls(['/chrome/elements/pgp_block.htm'], { sleep: 1 });
+      // but getFrame will fail, why?
+      const pgpBlockPage = await gmailPage.getFrame(['pgp_block.htm']);
```

Comment:
Ah, puppeteer bug https://github.com/puppeteer/puppeteer/issues/2548

The solution is to copy the url and open in separate frame. At which point.. you might as well make that a mock test where you hard code the url.. like other decrypt tests - unfortunately.

Reply:
> Ah, puppeteer bug puppeteer/puppeteer#2548

Thanks, subscribed :+1:

Reply:
> At which point.. you might as well make that a mock test where you hard code the url.. like other decrypt tests - unfortunately.

Yeah, that kinda ruins the point of the test. What do you think about using a visual test in this case? I'm talking about taking the screenshot of `iframe.pgp_block` and comparing it to the expected one. This is the result of `ElementHandle.screenshot`:

![22](https://user-images.githubusercontent.com/6059356/76171145-c9613f80-6190-11ea-9636-f406711687b1.png)

Reply:
That I think would be too too noisy - when we change css etc.

Reply:
> That I think would be too too noisy - when we change css etc.

Yeah, agree.